releases/release-4.0.9.md

```diff
@@ -0,0 +1,122 @@
+    - Fix the bug that `trace-region-flow` will be accidentally changed to `false`. [#3120](https://github.com/pingcap/pd/pull/3120)
+
++ TiFlash
+
+    - Fix the problem that `INFORMATION_SCHEMA.CLUSTER_HARDWARE` may contain information about disks not in use. [#1248](https://github.com/pingcap/tics/pull/1248)
+    - Fix the issue that the memory consumption statistic of Delta Cache is smaller than actual usage. [#1212](https://github.com/pingcap/tics/pull/1212)
+
+## Others
+
++ TiDB
+
+    - Fix issue of explain for connection can't show the last query plan/ [#21315](https://github.com/pingcap/tidb/pull/21315)
+    - Add metric for connection idle time [#21301](https://github.com/pingcap/tidb/pull/21301)
+    - Fix the issue that when Index Merge is used in a transaction with RC isolation level, the result may be incorrect. [#21253](https://github.com/pingcap/tidb/pull/21253)
+    - Planner: fix wrong point get plan generation in fast plan code path [#21244](https://github.com/pingcap/tidb/pull/21244)
+    - Expression: keep the original data type when doing date arithmetic operations [#21176](https://github.com/pingcap/tidb/pull/21176)
+    - Ban (index) merge join heuristically when convert equal conditions to other conditions. [#21146](https://github.com/pingcap/tidb/pull/21146)
+    - Ddl: Fix partition table's partition column values type check [#21136](https://github.com/pingcap/tidb/pull/21136)
+    - Make dump binary time more compatible with MySQL [#21135](https://github.com/pingcap/tidb/pull/21135)
+    - Differentiate types for user variables [#21107](https://github.com/pingcap/tidb/pull/21107)
+    - Fix auto-id allocation failed because of transaction's write-conflict retry. [#21079](https://github.com/pingcap/tidb/pull/21079)
+    - Fix The JSON Data can not import to TiDB correctly by `load data` [#21074](https://github.com/pingcap/tidb/pull/21074)
+    - Fix up displaying executor runtime info & Add runtime information for DML in explain analyze [#21066](https://github.com/pingcap/tidb/pull/21066)
+    - Deprecate `enable-streaming`. [#21055](https://github.com/pingcap/tidb/pull/21055)
+    - Avoid unnecessary warnings/errors when folding constants in shortcut-able expressions [#21040](https://github.com/pingcap/tidb/pull/21040)
+    - TiDB now provides an error for statements that use the syntax `LOCK IN SHARE MODE`. This helps alert users that this feature is not implemented by TiDB, but a workaround to restore the previous behavior is available by setting `tidb_enable_noop_functions=1`. [#21005](https://github.com/pingcap/tidb/pull/21005)
+    - Set correct default value for new added enum column. [#20998](https://github.com/pingcap/tidb/pull/20998)
+    - Allow set GOGC in config file. [#20922](https://github.com/pingcap/tidb/pull/20922)
+
++ TiKV
+
+    - Fix a bug that Coprocessor may return wrong result when there are > 255 columns. [#9131](https://github.com/tikv/tikv/pull/9131)
```

Comment:
All the items that starts with `Fix` should probably be moved to bug fix.

Reply:
moved